### PR TITLE
EZP-30985: Removed usage of deprecated AdvancedUserInterface

### PR DESCRIPTION
**Pull request by emodric**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30985](https://jira.ez.no/browse/EZP-30985)
| **Required by** | ezsystems/ezplatform#469
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `master`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

### QA

- [x] Sanity checks for password expiration,
- [x] Sanity checks for user account locking/disabling,
- [x] Sanity checks for REST login endpoint,
- [x] Run Behat suite with this branch and ezsystems/ezplatform#469.

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

**Comment by emodric** [comment]
Something like this? 11db3ecb11a561e74b8e24345ece882c9deb8324

File: eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        $apiUser = $user->getAPIUser();
+
+        if (!$apiUser->enabled) {
+            throw new DisabledException();
+        }
+
+        if ($this->userService->getPasswordInfo($apiUser)->isPasswordExpired()) {
+            throw new CredentialsExpiredException();
```
Comment (adamwojs) [inline eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php]: User credentials expiration should be checked in `checkPostAuth` method. Here is a good example: https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Security/Core/User/UserChecker.php

File: eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+namespace eZ\Publish\Core\MVC\Symfony\Security;
+
+use eZ\Publish\API\Repository\UserService;
+use Symfony\Component\Security\Core\Exception\CredentialsExpiredException;
+use Symfony\Component\Security\Core\Exception\DisabledException;
+use Symfony\Component\Security\Core\User\UserCheckerInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+use eZ\Publish\Core\MVC\Symfony\Security\UserInterface as EzUserInterface;
+
+final class UserChecker implements UserCheckerInterface
```
Comment (adamwojs) [inline eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php]: User Checker needs to be enabled in the firewall configuration. Ref. https://symfony.com/doc/current/security/user_checkers.html#enabling-the-custom-user-checker

File: eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
```
Comment (adamwojs) [inline eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php]:
```suggestion

/**
 * @copyright Copyright (C) eZ Systems AS. All rights reserved.
 * @license For full copyright and license information view LICENSE file distributed with this source code.
 */
declare(strict_types=1);
```

File: eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php (hunk 2 above)
Comment (emodric) [reply eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php]: Yep. I can do it now, or after this PR is approved, you can choose :)

File: eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php (hunk 1 above)
Comment (emodric) [reply eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php]: Done.

File: eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php (hunk 2 above)
Comment (emodric) [reply eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php]: I'm currently at a conference, so give me an hour or two :)

Comment (adamwojs) [reply eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php]: No problem, take your time ;-)

Comment (emodric) [reply eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php]: PR @ https://github.com/ezsystems/ezplatform/pull/469

Comment (adamwojs) [reply eZ/Publish/Core/MVC/Symfony/Security/UserChecker.php]: Thnak you ;-)

**Comment by micszo** [comment]
With these changes the message `User account is locked.` is not displayed on login page. 

**Steps**:
1. Edit an existing user and set Enabled to false.
2. Try to log in to `/admin` with this user.

**Actual**: Login page reloads, no message is displayed.
**Expected**: `User account is locked.` message is displayed.

**Comment by emodric** [comment]
@micszo Alright, it turns out exceptions need a message, they won't use a fallback message defined in the exception itself.

Can you tell me what is the expected error message for expired passwords, so I can update?

**Comment by micszo** [comment]
For expired passwords it is:
`Your password has expired`
`For security reasons, your password has expired and needs to be changed. An email has been sent to you with instructions.`
(in credentials_expired.html.twig)
@adamwojs please confirm that is the correct message.

**Comment by emodric** [comment]
@micszo How do I specify when the password expires?

**Comment by micszo** [comment]
On the level of Content Type in the ezuser field.

**Comment by emodric** [comment]
Hm. I get the error message `Template "@ezdesign/Security/error/credentials_expired.html.twig" is not defined.`

**Comment by emodric** [comment]
Admin PR @ https://github.com/ezsystems/ezplatform-admin-ui/pull/1109

**Comment by micszo** [comment]
`User account is locked.` msg is back now. ✅

**Comment by micszo** [comment]
This bug blocks completing the test scope: https://jira.ez.no/browse/EZP-31012 \cc @adamwojs

**Comment by emodric** [comment]
Hah, yes, I've tested this with manually updated `password_updated_at` column in the database :D

**Comment by micszo** [comment]
Indeed, that does the trick! (:

**Comment by micszo** [comment]
Testing discovered: https://jira.ez.no/browse/EZP-31051

**Pull request merged**